webpack/index.js

```diff
@@ -0,0 +1,22 @@
+import URI from 'urijs';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { mount } from 'foremanReact/common/MountingService';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import componentRegistry from 'foremanReact/components/componentRegistry';
+import JobInvocationContainer from './react_app/components/job_invocations';
+import './react_app/redux/reducers';
+
+componentRegistry.register({
+  name: 'JobInvocationContainer',
+  type: JobInvocationContainer,
+});
+
+if (window.location.href.match(/job_invocations/)) {
```

Comment:
What if turbolinks will navigate to the URL? will it still work?

In Foreman, we usually mount the components from inside the view file:
https://github.com/theforeman/foreman/blob/45d1359a8bb5d7ab75f1c0b83d28e8fff8eb8193/app/views/dashboard/_status_chart_widget.html.erb#L4-L6

I guess it will even be easier to get the `jobInvocationId` because you can get it from rails.

Reply:
Unfortunately I don't think that's possible to do in plugins. At the point when Rails loads the page, plugin components are not available yet, so we have to 'hack it'.

```
JobInvocationContainer among BookmarkContainer, PieChart, StatisticsChartsList, PowerStatus, NotificationContainer, ToastNotifications, StorageContainer, PasswordStrength
componentRegistry.js:45:12
 action TOASTS_CLEAR @ 11:19:40.599
```

Katello also needs to do this for the experimental UI component - https://github.com/Katello/katello/blob/master/webpack/index.js#L18

cc @waldenraines @danseethaler

Reply:
Turbolinks works with this